File: android/app/src/main/java/ae/findmybay/data/api/PaymentApi.kt
package ae.findmybay.data.api

import ae.findmybay.data.api.dto.PaymentIntentDto
import ae.findmybay.data.api.dto.PaymentStatusDto
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface PaymentApi {

    /** Create or fetch a payment intent for a pending_payment booking. */
    @POST("bookings/{id}/pay")
    suspend fun createIntent(@Path("id") id: String): PaymentIntentDto

    /** Poll the current state of a payment after returning from the Custom Tab. */
    @GET("payments/{ref}/status")
    suspend fun status(@Path("ref") ref: String): PaymentStatusDto
}

File: ios/FindMyBay/Data/API/PaymentAPI.swift
// PaymentAPI.swift
//
// Payment endpoints — parity with `data/api/PaymentApi.kt`.

import Foundation

protocol PaymentAPI {
    func createIntent(bookingId: String) async throws -> PaymentIntentDTO
    func status(ref: String) async throws -> PaymentStatusDTO
}

struct PaymentAPIImpl: PaymentAPI {
    let client: APIClient

    func createIntent(bookingId: String) async throws -> PaymentIntentDTO {
        try await client.request(.POST, path: "bookings/\(bookingId)/pay")
    }

    func status(ref: String) async throws -> PaymentStatusDTO {
        try await client.request(.GET, path: "payments/\(ref)/status")
    }
}
